src/releaseNotesProvider.ts

```diff
@@ -0,0 +1,243 @@
+import { env, ExtensionContext } from "vscode";
+import * as vscode from "vscode";
+import * as semver from "semver";
+import { Remarkable } from "remarkable";
+import { fetchFrom } from "./util";
+import { Either, makeLeft, makeRight } from "./types";
+
+const versionKey = "metals-server-version";
+
+/**
+ * Show release notes if possible, swallow errors since its not a crucial feature.
+ * Treats snapshot versions like 0.11.6+67-926ec9a3-SNAPSHOT as a 0.11.6.
+ */
+export async function showReleaseNotesIfNeeded(
+  context: ExtensionContext,
+  serverVersion: string,
+  outputChannel: vscode.OutputChannel
+) {
+  try {
+    // context.globalState.update(versionKey, "0.11.5");
+    const result = await showReleaseNotes(context, serverVersion);
+    if (result.kind === "left") {
+      const msg = `Release notes was not shown: ${result.value}`;
+      outputChannel.appendLine(msg);
+    }
```

Comment:
I've decided to introduce Either like datatype to handle errors/logging. I think that logging one line for every vscode startup won't be pain for users (although some of them are very curious and may ask what are release notes :D). But in case of any errors or problems this may be helpful. Probably testing on wsl/remote container will be easier either.